### DISCUSSION TITLE
How to find Sheetview and DrawingView from bis

### DISCUSSION THREAD

**Discussion by myl-Giser** [iTwin/itwinjs-core#5396] [Q&A]
One dgn file references two files,I want to know which file the 2D SheetView and DrawingView come from, and how to find the corresponding relationship from bis?

**Comment by pratikshan85** [comment]
@myl-Giser Can you please provide us the dgn files to investigate the issue in more detail?

**Comment by myl-Giser** [comment]
This is my dgn file.
[dgn.zip](https://github.com/iTwin/itwinjs-core/files/11248287/dgn.zip)

**Reply by navneetk897** [reply]
@myl-Giser The relationship between 2D sheetview, drawingview and file is through its ElementAspect properties.

Example: ```BisCore.DrawingViewDefinition``` and  ```BisCore.ExternalSourceAspect``` is related through ```BisCore.ElementOwnsMultiAspects```.
The ```BisCore.ExternalSourceAspect``` contain property called scope which have EcInstanceId for the ```BisCore.RepositoryLink``` which have the information about the file.

Please check the below link for more information about ElementAspect.
https://www.itwinjs.org/bis/guide/fundamentals/elementaspect-fundamentals/

This is the query that will get you EcInstanceId for the ```BisCore.RepositoryLink```.
```
SELECT multiAspect.scope.id RepositoryId
FROM bis.DrawingViewDefinition drawing
	JOIN bis.ElementOwnsMultiAspects multiAspectRel ON multiAspectRel.SourceECInstanceId = drawing.EcInstanceId
	JOIN bis.ExternalSourceAspect multiAspect ON multiaspect.ECInstanceId = multiAspectRel.EcInstanceId
WHERE drawing.ECInstanceId IN (
		SELECT ECInstanceId
		FROM bis.DrawingViewDefinition
		LIMIT 1
)
```
Above query is for ```bis.DrawingViewDefinition``` . If you want query for ```bis.SheetViewDefinition``` you can change the above query accordingly. And in the subquery LIMIT is set to 1 which is just taking the first element of ```bis.DrawingViewDefinition``` you can also change that, if you want this query to run for multiple elements.

Output of the above query.

![image](https://user-images.githubusercontent.com/67821826/232681659-4c08012a-4f87-43c1-8c9a-94fad59d0b14.png)



Then you have to query the query given below.

```select userLabel from bis.Element where ECInstanceId=0x20000000004```

Output of the above query.

![image](https://user-images.githubusercontent.com/67821826/232681870-14ec0d44-2721-4db3-b5d9-8c830d30c298.png)

**Reply by myl-Giser** [reply]
Thanks @navneetk897,@pratikshan85
